src/components/listBox/ListBox.tsx

```diff
@@ -3,28 +3,59 @@ import * as React from 'react';
 import * as _ from 'underscore';
 import { IItemBoxProps, ItemBox } from '../itemBox/ItemBox';
 
-export interface IListBoxProps {
+export interface IListBoxOwnProps {
   items: IItemBoxProps[];
-  onOptionClick?: (option: IItemBoxProps) => void;
   noResultItem?: IItemBoxProps;
   classes?: string[];
+  id?: string;
+  multi?: boolean;
+}
+
+export interface IListBoxStateProps {
+  selected?: string[];
 }
 
-export class ListBox extends React.Component<IListBoxProps> {
+export interface IListBoxDispatchProps {
+  onRender?: () => void;
+  onDestroy?: () => void;
+  onOptionClick?: (option: IItemBoxProps) => void;
+}
+
+export interface IListBoxProps extends IListBoxOwnProps, IListBoxStateProps, IListBoxDispatchProps {}
+
+export class ListBox extends React.Component<IListBoxProps, {}> {
 
   static defaultProps: Partial<IListBoxProps> = {
     noResultItem: {
       value: 'No Items',
     },
   };
 
+  componentDidMount() {
```

Comment:
I like to use `componentWillMount` for this purpose. This way there can be some changes before rendering.